### PR TITLE
[Merged by Bors] - feat: port Data.Int.Bitwise

### PR DESCRIPTION
**Pull request by joneugster**
This seems to be ready for review and merge now. There's only one question about renaming `shiftl_coe_nat` and `shiftr_coe_nat` left.

### CODE REVIEW

File: Mathlib/Init/Data/Int/Bitwise.lean [hunk 1]
```diff
@@ -0,0 +1,119 @@
+/-
+Copyright (c) 2017 Mario Carneiro. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro
+
+! Corresponding mathlib SHA: 6c48d300da03eaa8e374eb31933001090917202f
```
[the remark this answers is not in the document]
Comment (rwbarton) [reply Mathlib/Init/Data/Int/Bitwise.lean:6]: It should be as at https://github.com/leanprover-community/lean3port/blob/master/Leanbin/Init/Data/Int/Bitwise.lean

[the remark this answers is not in the document]
Comment (joneugster) [reply Mathlib/Init/Data/Int/Bitwise.lean]: thx! done

**Comment by rwbarton** [comment]
bors merge

**Comment by rwbarton** [comment]
Thanks. I cleaned up some proofs to avoid non-terminal simps.

**Pull request closed without merging**